tests/unit/s2n_session_ticket_test.c

```diff
@@ -1051,23 +1051,45 @@ int main(int argc, char **argv)
 
     /* Test s2n_connection_is_session_resumed */
     {
-        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        /* TLS1.2 */
+        {
+            struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(conn);
+            conn->actual_protocol_version = S2N_TLS12;
 
-        EXPECT_TRUE(s2n_connection_get_protocol_version(server_conn) == S2N_TLS12);
-        EXPECT_TRUE(server_conn->handshake.handshake_type == INITIAL);
-        EXPECT_FALSE(s2n_connection_is_session_resumed(server_conn));
+            conn->handshake.handshake_type = INITIAL;
+            EXPECT_FALSE(s2n_connection_is_session_resumed(conn));
 
-        server_conn->handshake.handshake_type = NEGOTIATED | WITH_SESSION_TICKET;
-        EXPECT_TRUE(s2n_connection_is_session_resumed(server_conn));
+            conn->handshake.handshake_type = NEGOTIATED | WITH_SESSION_TICKET;
+            EXPECT_TRUE(s2n_connection_is_session_resumed(conn));
 
-        server_conn->actual_protocol_version = S2N_TLS13;
-        server_conn->handshake.handshake_type = INITIAL;
-        EXPECT_FALSE(s2n_connection_is_session_resumed(server_conn));
+            /* Ignores PSK mode */
+            conn->psk_params.type = S2N_PSK_TYPE_EXTERNAL;
+            EXPECT_TRUE(s2n_connection_is_session_resumed(conn));
```

Comment:
This test is a bit confusing as it is not a valid use-case.  There is no PSK external type for TLS1.2.  if we are testing invalid use-case  maybe we should call that out in the comment on line 1054?

Reply:
There is no PSK external type for TLS1.2, but PSK mode can be set even if TLS1.2 is negotiated. Minimally, it's possibly if we fall back to TLS1.2.

However, PSK mode (and everything else in psk_params) is ignored in TLS1.2, which is what this tests.

Reply:
This makes me wonder if we should make s2n_connection_set_psk_mode  a  `> TLS1.3` api instead.

Reply:
This makes me wonder if we should make `s2n_connection_set_psk_mode` and `s2n_config_set_psk_mode`  `>= TLS1.3` version apis instead.

Reply:
Same for `s2n_connection_append_psk` and any other PSK api that operates on the connection object.

Reply:
We can't. We don't know before negotiating what version we'll ultimately use.